pkg/provider/hpa_test.go

```diff
@@ -31,7 +32,8 @@ func (c mockCollector) Interval() time.Duration {
 
 func TestUpdateHPAs(t *testing.T) {
 	value := resource.MustParse("1k")
-	hpa := &autoscalingv2.HorizontalPodAutoscaler{
+
+	hpa := &autoscalingv1.HorizontalPodAutoscaler{
```

Comment:
Should we not create 2 tests for each version one?

Reply:
I tried now to create an HPA v2beta2 object and then read it via the HPA v2beta1 api. However for the fakeClient this is not connected like it is in a real Kubernetes cluster, so this doesn't work unfortunately :(

Effectively with this change we are only supporting v2beta1 API because we don't make use of any of the v2beta2 features yet. I think we can not realistically support this without having two code paths.

Maybe it good enough to just support v2beta1 for now so it works on GKE and then we can address the bigger issue once we need v2beta2 support? WDYT? @szuecs @arjunrn